### ISSUE TITLE
onclose in modal behavior

### ISSUE THREAD

**Issue by AviHafner** [Flyer53/jsPanel4#125]
Hello Stefan

I like a  lot all the components of jspanel.
I use **tooltip** a lot.

We have talked in the past about which product I should choose to display tooltip,
And I chose the product that you wrote as a  **perfect product for displaying tooltip**.
I really like **hint** and use it a lot.
The **progressbar** is just perfect and gives a perfect user  experience.

**The Minority Problems**
1. **onclose** in **modal** requires to end the function  **return true**
Otherwise the result is not reasonable.
This behavior has begun in recent versions.

2. In your documentation on the page
[Get started with jsPanel 4]( https://jspanel.de/getstarted.html)
In your example of using css in cdn
the keyword   **rel="stylesheet"**
is Missing.

Examples of what I write about can be found in the url
http://www.admin.technion.ac.il/avi/

yours 
Avi Hafner

**Comment by Flyer53** [comment]
@AviHafner 
Hi Avi,
Always a pleasure Avi. And thanks for the kind words 😄 

**Your No 2:** thanks for the hint ... fixed it already

**Your No 1** is a bit more complicated:
First of all the behaviour is absolutely as I would expect it. Yes, really. But the docs about the modal extension are missing the appropriate note on that. I'll have to correct that.
_**Now the explanation.**_ If you go to the documentation of the `onclosed` callback and carefully read the part [Using an array of functions](https://jspanel.de/api.html#options/onclosed/using-array) and then ckeck the example code below (Example 2) you'll see that the first callback function has to return a truthy value in order to have the second callback function execute as well. But you added only one callback function to the modal, right! Well, yes, but the modal extension also adds a callback to `onclosed`, the one removing the modal backdrop. So a modal panel always has one more onclosed callback than you specify in the modal config. The one you added is the first, and the one removing the modal backdrop ends up as the second. So in one of your examples the first callback (the one you added) has no return value, which means falsy. And that means the second callback (the one removing the backdrop) is never executed.
Bottom line: Behaviour is fully correct. I just have to make a note on this in the modal documentation.

A better solution of course would be if the callback removing the modal backdrop would always be the first callback in queue. I'll see whether I can change this in the next update.

I hope I could explain it reasonablyy well.

Best regards and stay healthy,
Stefan

**Comment by AviHafner** [comment]
Stefan
Thanks you for an in-depth explanation.

**Another Problem that I can't solve** in a jsPanel **modal** window
After the jsPanel modal window opens, 
If the client presses on **enter key** the screen goes into a state I do not know how to solve.
the close stop to function, 
The client needs to press the close twice to close the panel . 

yours
Avi Hafner

**Comment by Flyer53** [comment]
@AviHafner 
Now that's a funny issue that actually applies to all panels, not just modals.
I assume your test page has some kind of button creating the modal on click. So what happens if you click this button: The handler for the button is executed (creating the modal) **AND** the button receives focus. Now you hit <kbd>enter</kbd> and since the button still has focus a second modal is created. You can confirm this with the dev tools checking the HTML.
How to solve this?  You could add a bit of code to the click handler that removes focus from the button. Example:
```javascript
// event listener for button with id 'testbutton'
document.getElementById('testbutton').addEventListener('click', function(e) {
    // create a modal jsPanel
    jsPanel.modal.create();
    // remove focus from clicked button
    e.target.closest('button').blur();
    // since e.target might point to a childelement inside the button element I use ...
    // ... e.target.closest('button') to get the button element itself
});
```
Another option that might be preferable is to give the modal a unique id and simply check whether an element with this id already exists when clicking the button. Example:
```javascript
document.getElementById('testbutton').addEventListener('click', function() {
    // create modal only if not there already
    if (!document.getElementById('modal-1')) {
        jsPanel.modal.create({
            id: 'modal-1'
        });
    }
});
```
Hope this helps. Yours,
Stefan

**Comment by AviHafner** [comment]
Stefan
a big thanks
**e.target.closest('a').blur();**
is a perfect solution that Improved the behavior of **enter key**

yours
Avi Hafner
